### PR TITLE
Feature/appeals 36234.prodtest

### PR DESCRIPTION
**Pull request by amybids**
Resolves [Jira Issue Title](https://jira.devops.va.gov/browse/JIRA-12345)

# Description
Please explain the changes you made here.

## Acceptance Criteria
- [ ] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Frontend
## User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

# Backend
## Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

## Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
  * [ ] Check that calls are wrapped in MetricService record block
* [ ] Check that all configuration is coming from ENV variables
  * [ ] Listed all new ENV variables in description
  * [ ] Worked with or notified System Team that new ENV variables need to be set
* [ ] Update Fakes
* [ ] For feature branches: Was this tested in Caseflow UAT

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [ ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

File: app/queries/appeals_non_priority_ready_for_distribution.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          # appeals = docket.ready_to_distribute_appeals.select { |appeal| appeal["aod"] == 0 }
+          # legacy_rows(appeals, docket, sym)
+        else
+          appeals = docket.ready_priority_nonpriority_appeals(priority: false, ready: true)
+          ama_rows(appeals, docket, sym)
+        end
+      end
+  end
+
+  # def self.legacy_rows(appeals, docket, sym)
```
Comment (craigrva) [inline app/queries/appeals_non_priority_ready_for_distribution.rb:54]: Should this commented code be removed?

File: client/app/testSeeds/pages/TestSeedsApp.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// import React from 'react';
```
Comment (craigrva) [inline client/app/testSeeds/pages/TestSeedsApp.jsx:1]: Commented code

**Review by craigrva — CHANGES_REQUESTED** [review]

File: app/controllers/legacy_tasks_controller.rb [hunk 3]
```diff
@@ -136,7 +136,13 @@ def validate_user_id
...
   def validate_user_role
-    return invalid_role_error unless ROLES.include?(user_role)
+    return true if ROLES.include?(user_role)
+
+    Rails.logger.info("User with roles #{current_user.roles.join(', ')} "\
+      "couldn't access #{request.original_url}: Error: #{invalid_role_error}")
+
+    session["return_to"] = request.original_url
+    redirect_to "/unauthorized"
   end
```
Comment (zurbergram) [inline app/controllers/legacy_tasks_controller.rb:146]: If no longer using `invalid_role_error` the file `app/controllers/concerns/errors.rb` should be deleted after confirming this is only instance of usage

File: app/controllers/concerns/errors.rb [hunk 4]
```diff
@@ -3,11 +3,12 @@
 module Errors
   extend ActiveSupport::Concern
   def invalid_role_error
```
Comment (zurbergram) [inline app/controllers/concerns/errors.rb:5]: Is this still being used?

File: app/controllers/test_docket_seeds_controller.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  #   return true if current_user&.organizations && current_user.organizations.any?(&:users_can_view_levers?)
+
+  #   session["return_to"] = request.original_url
+  #   redirect_to "/unauthorized"
+  # end
+
+  def check_environment
+    return true if Rails.env.development?
+    return true if Rails.deploy_env?(:demo)
+    return true if Rails.deploy_env?(:uat)
```
Comment (zurbergram) [inline app/controllers/test_docket_seeds_controller.rb:45]:
```suggestion
    return true if Rails.env.development? || Rails.deploy_env?(:demo) || Rails.deploy_env?(:uat)
```

File: app/controllers/test_seeds_controller.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+require "csv"
+
+class Test::TestSeedsController < ApplicationController
+  before_action :check_environment
+  before_action :verify_access, only: [:seeds]
+  before_action :authorize_admin, only: [:seeds]
+
+  def seeds
+    # seeds
+    render "/test/seeds"
```
Comment (zurbergram) [inline app/controllers/test_seeds_controller.rb:12]: This url name is too generic for what this page is doing, something like `/test/seeds/acd` would be more appropriate as a page like `/test/seeds/` would be a generic "run any seed file" or something similar

File: app/models/docket.rb [hunk 7]
```diff
@@ -9,6 +9,9 @@ def docket_type
     fail Caseflow::Error::MustImplementInSubclass
   end
 
+  PRIORITY = "priority"
+  NON_PRIORITY = "non_priority"
```
Comment (zurbergram) [inline app/models/docket.rb:13]: Should be declared before any method declarations

**Review by zurbergram — CHANGES_REQUESTED** [review]
This PR has numerous issues that need to be resolved before can be approved

File: app/controllers/test_docket_seeds_controller.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    RequestStore[:current_user] = current_user
+    DistributionTask.where(status: "assigned").map { |t| t.update!(status: "on_hold") }
+    VACOLS::Case.where(bfcurloc: %w[81 83]).map { |c| c.update!(bfcurloc: "testing") }
+
+    head :ok
+  end
+
+  private
+
+  # def verify_access ##future work
```
Comment (zurbergram) [inline app/controllers/test_docket_seeds_controller.rb:35]: Commented out code shouldn't be committed to release branches, we have git history if need to recover this

File: app/queries/appeals_non_priority_ready_for_distribution.rb [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
```
Comment (zurbergram) [inline app/queries/appeals_non_priority_ready_for_distribution.rb:54]: Commented out code shouldn't be committed to release branches, we have git history if need to recover this

**Review by zurbergram — CHANGES_REQUESTED** [review]
Commented out code shouldn't be committed to release branches, we have git history if need to recover this

Test Seeds URL is still showing in incorrect location and it's url and file directory need to be discussed before allowing the code into a release branch

**Review by zurbergram — APPROVED** [review]

**Pull request closed without merging**